Limit and LimitWeighted operator docs (#28394)
akka/akka: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limitWeighted.md

# limitWeighted

> Limit a max total weight for incoming elements

Reviewer: ```suggestion
Limit the total weight of incoming elements
```
?

@ref[Simple operators](../index.md#simple-operators)

## Signature

@apidoc[Flow.limitWeighted](Flow) { scala="#limitWeighted[T](max:Long)(costFn:Out=&gt;Long):FlowOps.this.Repr[Out]" java="#limitWeighted(long,akka.japi.function.Function)" } 

## Description

A weight function returns the weight of each element, then the total accumulated weight  is compared to a max and if it has passed the max the stream is failed with a @apidoc[StreamLimitReachedException](StreamLimitReachedException).

> See also @ref:[limit](limit.md) which puts a number on the number of elements instead (the same as always returning `1` from the weight function).

Reviewer 2: ```suggestion
See also @ref:[limit](limit.md) which puts a limit on the number of elements instead (the same as always returning `1` from the weight function).
```

## Examples

> `limitWeighted` can protect a stream coming from an untrusted source into an in memory aggregate that grows with the number of element from filling the heap and causing an OOM.

Reviewer 2: ```suggestion
`limitWeighted` can protect a stream coming from an untrusted source into an in-memory aggregate that grows with the number of elements from filling the heap and causing an out-of-memory error.
```
